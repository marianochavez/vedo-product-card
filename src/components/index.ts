import { ProductButtons } from './ProductButtons';
import { ProductCard as ProductCardHOC } from './ProductCard';
import { ProductImage } from './ProductImage';
import { ProductTitle } from './ProductTitle';
import { ProductCardHOCProps } from '../interfaces/interfaces';

export { ProductImage } from './ProductImage';
export { ProductTitle } from './ProductTitle';
export { ProductButtons } from "./ProductButtons";


// Para poder llamar ProductCard.Title, ProductCard.Image, ProductCard.Buttons
export const ProductCard: ProductCardHOCProps = Object.assign(ProductCardHOC, {
    Title: ProductTitle,
    Image: ProductImage,
    Buttons: ProductButtons,
});

export default ProductCard;
