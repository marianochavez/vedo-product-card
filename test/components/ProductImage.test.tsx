import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard, ProductImage } from '../../src/components';
import { product2 } from '../data/products';


describe('ProductImage', () => {
    test('debe mostrar el componente correctamente con la imagen personalizada', () => {
        const wrapper = renderer.create(
            <ProductImage img='https://via.placeholder.com/150' />
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('debe mostrar componente correctamente con la imagen de producto 1', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => <ProductImage />
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    })
})