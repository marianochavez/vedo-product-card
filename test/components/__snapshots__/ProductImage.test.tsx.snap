// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe mostrar componente correctamente con la imagen de producto 1 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage debe mostrar el componente correctamente con la imagen personalizada 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="https://via.placeholder.com/150"
/>
`;
